posthog/hogql/bytecode.py

```diff
@@ -105,6 +105,18 @@ def visit_constant(self, node: ast.Constant):
             raise NotImplementedException(f"Constant type `{type(node.value)}` is not supported")
 
     def visit_call(self, node: ast.Call):
+        if node.name == "not" and len(node.args) == 1:
+            return [*self.visit(node.args[0]), Operation.NOT]
+        if node.name == "and" and len(node.args) > 1:
+            args = []
+            for arg in reversed(node.args):
+                args.extend(self.visit(arg))
+            return [*args, Operation.AND, len(node.args)]
+        if node.name == "or" and len(node.args) > 1:
+            args = []
+            for arg in reversed(node.args):
+                args.extend(self.visit(arg))
+            return [*args, Operation.OR, len(node.args)]
```

Comment:
I'm a bit ignorant on why this bytecode is designed in this particular way: curious why is the operator _after_ the args?

Reply:
We add the constants to the stack as we encounter them, and then take `len(node.args)` of them from the stack to do the OR comparison.